### PR TITLE
N31: Vach

### PR DESCRIPTION
**Pull request by zeskeertwee**
Here's my and @sokorototo's contribution to the newsletter!

*Part of #949*

### CODE REVIEW

**Comment by zeskeertwee** [comment]
*this is a draft*

File: content/news/031/index.md [hunk 1]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
 
 ## Library Updates
 
+### [vach]
+
+[vach] is an archiving file format developed from the ground up for games and other realtime applications by [@zeskeertwee] and [@sokorototo]. It is written in pure Rust.
+
+It's primary objectives (in no perticular order) are:
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:83]:
```suggestion
Its primary objectives (in no particular order) are:
```

File: content/news/031/index.md [hunk 2]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+
+- Have a simple, yet complete API.
+- Fine control over each individual entry in an archive, ie neighbouring entries can have vastly different compression schemes.
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:86]:
```suggestion
- Fine control over each individual entry in an archive, i.e. neighbouring entries can have vastly different compression schemes.
```

File: content/news/031/index.md [hunk 3]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+- Support compression (with multiple compression schemes), enryption, signatures for data authentication
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:87]:
```suggestion
- Support compression (with multiple compression schemes), encryption, signatures for data authentication
```

File: content/news/031/index.md [hunk 4]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+- Efficient fetching of data by avoiding unnecessary traversal of the file. Once a file is parsed locations of every entry are known to the loader.
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:88]:
```suggestion
- Efficient fetching of data by avoiding unnecessary traversal of the file. Once a file is parsed, locations of every entry are known to the loader.
```

File: content/news/031/index.md [hunk 5]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+- Be as compact as possible, the smallest valid archive is only 13 bytes.
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:89]:
```suggestion
- Be as compact as possible - the smallest valid archive is only 13 bytes.
```

File: content/news/031/index.md [hunk 6]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+- Each entry has some metadata attached to it, this is implemented using bitflags and up to 8 bits are free to the user.
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:90]:
```suggestion
- Each entry has some metadata attached to it - this is implemented using bitflags and up to 8 bits are free to the user.
```

File: content/news/031/index.md [hunk 7]
```diff
@@ -76,6 +76,29 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+- Has mutlithreaded implementations of both the loader and the writer.
+
+A [cli][vach-cli] is allowing one to use vach as a general purpose archive format. The CLI is fully multithreaded, allowing for insane un/packing speeds.
```
Comment (17cupsofcoffee) [inline content/news/031/index.md:93]:
```suggestion
A [CLI][vach-cli] is allowing one to use vach as a general purpose archive format. The CLI is fully multithreaded, allowing for insane un/packing speeds.
```

**Review by 17cupsofcoffee — COMMENTED** [review]
The content looks good (made some minor tweaks). Just needs reformatting to make the linter happy.

File: content/news/031/index.md [hunk 8]
```diff
@@ -76,6 +76,37 @@ If needed, a section can be split into subsections with a "------" delimiter.
...
+other realtime applications by [@zeskeertwee] and [@sokorototo]. It is
+written in pure Rust.
+
+Its primary objectives (in no particular order) are:
+
+- Have a simple, yet complete API.
+- Fine control over each individual entry in an archive, i.e. neighbouring
+  entries can have vastly different compression schemes.
+- Support compression (with multiple compression schemes), encryption,
+  signatures for data authentication
```
Comment (AngelOnFira) [inline content/news/031/index.md:91]:
```suggestion
- Support compression (with multiple compression schemes), encryption,
  and signatures for data authentication
```

**Review by AngelOnFira — APPROVED** [review]

**Pull request merged**